docs/static/env-vars.asciidoc

```diff
@@ -12,6 +12,7 @@ environment variable is undefined.
 * You can add environment variable references in any plugin option type: string, number, boolean, array, or hash.
 * Environment variables for list-type URI parameters can support lists of space-delimited values. Currently, other non-URI based options do not support lists of values. See <<plugin-concepts>>
 * Environment variables are immutable. If you update the environment variable, you'll have to restart Logstash to pick up the updated value.
+* Environment variables references in `config.string` comments are not allowed. Remove the `$` sign to avoid pipeline config failures.
```

Comment:
I agree with the sentiment from @donoghuc where we should be clear about the actual behaviour, while still providing guidance:

```suggestion
* Environment variables references in `config.string` comments will still be evaluated due to configuration parsing limitations and therefore discouraged. Remove the `$` sign to avoid potential pipeline initialization failures.
```